routemodel/data_retrieval/elevations.py

```diff
@@ -0,0 +1,165 @@
+import sys
+import os.path
+from config import API_KEY, BASE_URL
+
+sys.path.append(os.path.dirname(__file__))
+
+# Dict to convert Base 10 to Bing's Base 64 compressed text used in points_builder function
+ENCODER_DICT = {0: "A", 1: "B", 2: "C", 3: "D", 4: "E", 5: "F", 6: "G", 7: "H",
+                8: "I", 9: "J", 10: "K", 11: "L", 12: "M", 13: "N", 14: "O",
+                15: "P", 16: "Q", 17: "R", 18: "S", 19: "T", 20: "U", 21: "V",
+                22: "W", 23: "X", 24: "Y", 25: "Z", 26: "a", 27: "b", 28: "c",
+                29: "d", 30: "e", 31: "f", 32: "g", 33: "h", 34: "i", 35: "j",
+                36: "k", 37: "l", 38: "m", 39: "n", 40: "o", 41: "p", 42: "q",
+                43: "r", 44: "s", 45: "t", 46: "u", 47: "v", 48: "w", 49: "x",
+                50: "y", 51: "z", 52: "0", 53: "1", 54: "2", 55: "3", 56: "4",
+                57: "5", 58: "6", 59: "7", 60: "8", 61: "9", 62: "_", 63: "-"}
+
+
+def points_builder(coordinates: list):
+    """
+    Compresses points to compressed string
+    @param coordinates: list of dictionaries of coordinates Ex: [{lat1: long1}, {lat2: long2},... {latN: longN}]
+    @return: String of encoded points for sending to API  Ex: 'vx1vilihnM6hR7mEl2Q'
+    Algorithm taken from:
+    https://docs.microsoft.com/en-us/bingmaps/rest-services/elevations/point-compression-algorithm
+    """
+    compressed_points = ""
+    lat = 0
+    long = 0
+    # Step 1: Start with a set of latitude and longitude values.
+    for index, pair in enumerate(coordinates):
+        for key in pair.keys():
+            # Step 2: Multiply each value by 100000 and round each result to the nearest int
+            new_lat = int(round(float(key) * 100000))
+            new_long = int(round(float(pair[key]) * 100000))
+
+            # Step 3: Calculate the difference between every pair of values
+            dx = (new_lat - lat)
+            dy = (new_long - long)
+            lat = new_lat
+            long = new_long
+
+            # Step 4: Multiply each value by 2
+            dy *= 2
+            dx *= 2
+
+            # Step 5: for negative change it to be a positive value, and then subtract 1
+            if dx < 0:
+                dx = abs(dx) - 1
+            if dy < 0:
+                dy = abs(dy) - 1
+
+            # Step 6: For each pair of latitude and longitude coordinates compute
+            #         ((latitude + longitude) * (latitude + longitude + 1) / 2) + latitude
+            index = int(((dx + dy) * (dx + dy + 1) / 2) + dx)
+
+            # Step 7: For each number, form a list of numbers by dividing the number by 32 repeatedly and recording
+            #         each remainder, stop when the quotient reaches zero
+            rem = []
+            while index > 0:
+                rem.append(int(index % 32))
+                index = int(index / 32)
+
+            # Step 8: Add 32 to each number except for the last number in each list.
+            for i in range(0, len(rem) - 1):
```

Comment:
python thing ->  can use something like

```suggestion
rem = [i + 32 for i in rem]
```

Reply:
Sorry I don't know how to use tabs in comments

Reply:
Resolved this over slack